services/core-web/src/components/mine/Permit/ComparePermitConditionHistoryModal.tsx

```diff
@@ -0,0 +1,71 @@
+import React, { FC } from "react";
+import { Col, Row, Typography } from "antd";
+import { IMineReportPermitRequirement, IPermitAmendment, IPermitCondition } from "@mds/common/interfaces";
+import FormWrapper from "@mds/common/components/forms/FormWrapper";
+import ConditionDiffViewer from "./ConditionDiffViewer";
+import PermitConditionReportRequirements from "./PermitConditionReportRequirements";
+import { getMineReportPermitRequirements, getMineReportPermitRequirementsByAmendment } from "@mds/common/redux/selectors/permitSelectors";
+import { useSelector } from "react-redux";
+import { getConditionsWithRequirements } from "@mds/common/utils/helpers";
+
+export interface ComparePermitConditionHistoryModalProps {
+    currentAmendmentCondition: IPermitCondition;
+    previousAmendmentCondition: IPermitCondition;
+    mineGuid: string;
+    permitGuid: string;
+    latestAmendment: IPermitAmendment;
+    previousAmendment?: IPermitAmendment;
+}
+
+/**
+ * View that compares the current permit condition (including children conditions) with the previous permit condition
+ * in a diff viewer. Also displays the reports associated with the current and previous permit conditions.
+ */
+const ComparePermitConditionHistoryModal: FC<ComparePermitConditionHistoryModalProps> = (props) => {
+
+    const newMineReportPermitRequirements: IMineReportPermitRequirement[] = useSelector(
+        getMineReportPermitRequirements(props.permitGuid)
+    );
+
+    const previousMineReportPermitRequirements: IMineReportPermitRequirement[] = useSelector(
+        getMineReportPermitRequirementsByAmendment(props.permitGuid, props.previousAmendment?.permit_amendment_guid)
+    );
+
+    const oldReports = getConditionsWithRequirements([props.currentAmendmentCondition], previousMineReportPermitRequirements);
+    const newReports = getConditionsWithRequirements([props.currentAmendmentCondition], newMineReportPermitRequirements);
+
+    return (
+        <FormWrapper name={"test"} isModal onSubmit={() => { }}>
```

Comment:
I am a little bit wondering about these parameters and the empty div with `.ant-modal-footer`

Reply:
Oh wops, will update the name, and maybe make `onSubmit` optional on the `FormWrapper` (currently required). The `PermitConditionReportRequirements` have nested form fields far down in the tree, so the `FormWrapper` was needed.